netlify/next-runtime
# feat: add support for Next 12.3 middleware matchers

### Summary

Adds support for 12.3 middleware matchers, including `has` and `locale` support.

### Test plan

- Check that [the middleware demo](https://deploy-preview-1612--next-plugin-edge-middleware.netlify.app/conditional) works
- Load https://deploy-preview-1612--next-plugin-edge-middleware.netlify.app/conditional and check that that headers `x-middleware-next` and `x-is-deno` are *not* set
- Load the same page, but setting the request header `x-my-header` to `my-value`. Check that headers `x-middleware-next` and `x-is-deno` *are* set
- Load https://deploy-preview-1612--next-plugin-edge-middleware.netlify.app/_next/data/build-id/static.json and ensure that headers `x-middleware-next` and `x-is-deno` *are* set.

### Relevant links (GitHub issues, Notion docs, etc.) or a picture of cute animal

Fixes #1602. FIxes #1481. Fixes https://github.com/netlify/pod-ecosystem-frameworks/issues/230

![capy and duck](https://user-images.githubusercontent.com/213306/189306282-d0e54e2e-62d7-4194-88eb-e21dd5c3617e.png)

### Standard checks:




---

🧪 Once merged, make sure to update the version if needed and that it was published correctly.
